scripts/prepdocs.py

```diff
@@ -143,6 +149,65 @@ async def setup_file_strategy(credential: AsyncTokenCredential, args: Any) -> Fi
     )
 
 
+async def setup_intvectorizer_strategy(credential: AsyncTokenCredential, args: Any) -> Strategy:
+    storage_creds = credential if is_key_empty(args.storagekey) else args.storagekey
+    blob_manager = BlobManager(
+        endpoint=f"https://{args.storageaccount}.blob.core.windows.net",
+        container=args.container,
+        account=args.storageaccount,
+        credential=storage_creds,
+        resourceGroup=args.storageresourcegroup,
+        subscriptionId=args.subscriptionId,
+        store_page_images=args.searchimages,
+        verbose=args.verbose,
+    )
+
+    use_vectors = not args.novectors
+    embeddings: Union[AzureOpenAIEmbeddingService, None] = None
+    if use_vectors and args.openaihost != "openai":
+        azure_open_ai_credential: Union[AsyncTokenCredential, AzureKeyCredential] = (
+            credential if is_key_empty(args.openaikey) else AzureKeyCredential(args.openaikey)
+        )
+        embeddings = AzureOpenAIEmbeddingService(
+            open_ai_service=args.openaiservice,
+            open_ai_deployment=args.openaideployment,
+            open_ai_model_name=args.openaimodelname,
+            credential=azure_open_ai_credential,
+            disable_batch=args.disablebatchvectors,
+            verbose=args.verbose,
+        )
+
+    print("Processing files...")
+    list_file_strategy: ListFileStrategy
+    if args.datalakestorageaccount:
+        adls_gen2_creds = credential if is_key_empty(args.datalakekey) else args.datalakekey
+        print(f"Using Data Lake Gen2 Storage Account {args.datalakestorageaccount}")
+        list_file_strategy = ADLSGen2ListFileStrategy(
+            data_lake_storage_account=args.datalakestorageaccount,
+            data_lake_filesystem=args.datalakefilesystem,
+            data_lake_path=args.datalakepath,
+            credential=adls_gen2_creds,
+            verbose=args.verbose,
+        )
+    else:
+        print(f"Using local files in {args.files}")
+        list_file_strategy = LocalListFileStrategy(path_pattern=args.files, verbose=args.verbose)
+
+    document_action = DocumentAction.Add
```

Comment:
is remove/removeall not supported?

Reply:
Updated. Note once we remove document from blob, and run indexer, it ensures to remove all documents from search.